torch/distributed/_tensor/placement_types.py

```diff
@@ -251,15 +251,8 @@ def _to_replicate_tensor(
 
         # Unpad the tensor if the input tensor was padded
         if is_padded:
-            gathered_list = torch.chunk(result, num_chunks, dim=self.dim)
-            gathered_list = [
-                self._unpad_tensor(gathered_tensor, pad_size)  # type: ignore[misc]
-                if pad_size > 0
-                else gathered_tensor
-                for gathered_tensor, pad_size in zip(gathered_list, pad_sizes)
-            ]
-
-            result = torch.cat(gathered_list, dim=self.dim)
+            full_pad_size = sum(pad_sizes)
+            result = self._unpad_tensor(result, full_pad_size)
```

Comment:
Do we have a unit test to capture this? I mean since this is a fix so I guess some test cases might be failing?

Reply:
> Do we have a unit test to capture this? I mean since this is a fix so I guess some test cases might be failing?

I believe this is more of a better engineering so no test case is failing. So instead of chunking to create the tensor list and unpad each tensor, Wanchao's fix directly just unpad the all_gather result directly.

Reply:
Yep it's more like a improvement rather than a fix, there're already existing test capture these so I didn't add new tests. Updated the title to reflect